central/processlisteningonport/store/postgres/store.go

```diff
@@ -0,0 +1,748 @@
+// Code generated by pg-bindings generator.
+
+// XXX: There are custom bits applied on top of the generated code. Everything
+// what comes together with GetProcessListeningOnPort and
+// GetProcessListeningOnPortByNamespace is manually written.
+
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/logging"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/sync"
+	"gorm.io/gorm"
+)
+
+const (
+	baseTable = "process_listening_on_ports"
+
+	batchAfter = 100
+
+	// using copyFrom, we may not even want to batch.  It would probably be simpler
+	// to deal with failures if we just sent it all.  Something to think about as we
+	// proceed and move into more e2e and larger performance testing
+	batchSize = 10000
+
+	cursorBatchSize = 50
+
+	deleteBatchSize = 5000
+
+	// SQL query to join process_listening_on_port together with
+	// process_indicators. Used to provide information for queries like 'give
+	// me all PLOP by this deployment'.
+	// XXX: Verify the query plan to make sure needed indexes are in use.
+	getByDeploymentStmt = "SELECT plop.id, plop.serialized, " +
+		"proc.podid, proc.containername, " +
+		"proc.signal_containerid, " +
+		"proc.signal_name, proc.signal_args, proc.signal_execfilepath, " +
+		"proc.clusterid, proc.serialized as proc_serialized " +
+		"FROM process_listening_on_ports plop " +
+		"JOIN process_indicators proc " +
+		"ON plop.processindicatorid = proc.id " +
+		"WHERE proc.deploymentid = $1 AND plop.closed = false"
+)
+
+var (
+	log            = logging.LoggerForModule()
+	schema         = pkgSchema.ProcessListeningOnPortsSchema
+	targetResource = resources.DeploymentExtension
+)
+
+type Store interface {
```

Comment:
nit:  #4210 reordered the routines into a logical order.  Up to you if you want to mimic that.

Reply:
We have generated those bits again to apply the new ordering coming from this change.